### PR TITLE
Use http helper

### PR DESCRIPTION
**Pull request by ganmacs**
This PR should be merged after the release which includes https://github.com/fluent/fluentd/pull/2787.

### CODE REVIEW

**Comment by ganmacs** [comment]
ref https://github.com/fluent/fluent-plugin-prometheus/issues/48

File: lib/fluent/plugin/in_prometheus.rb [hunk 1]
```diff
@@ -66,98 +63,89 @@ def multi_workers_ready?
...
-        res.status = 500
-        res['Content-Type'] = 'text/plain'
-        res.body = $!.to_s
+      [200, { 'Content-Type' => ::Prometheus::Client::Formats::Text::CONTENT_TYPE }, full_result.get_metrics]
+    rescue => e
+      [500, { 'Content-Type' => 'text/plain' }, e.to_s]
+    end
+
+    def send_request_to_each_worker
+      bind == '0.0.0.0' ? '127.0.0.1' : @bind
```
Comment (repeatedly) [inline lib/fluent/plugin/in_prometheus.rb]: `bind =` is missing?

File: spec/fluent/plugin/in_prometheus_spec.rb [hunk 2]
```diff
@@ -47,6 +47,95 @@
     end
   end
 
+  describe '#start' do
```
Comment (repeatedly) [inline spec/fluent/plugin/in_prometheus_spec.rb]: No test for `<transport tls>` ?

File: spec/fluent/plugin/in_prometheus_spec.rb [hunk 3]
```diff
@@ -47,6 +47,95 @@
...
+    context 'when extra_conf is used' do
+      let(:config) do
+        %[
+          type prometheus
```
Comment (repeatedly) [inline spec/fluent/plugin/in_prometheus_spec.rb]: use `@type`

File: spec/fluent/plugin/in_prometheus_spec.rb [hunk 4]
```diff
@@ -47,6 +47,95 @@
...
+                                     logger: anything,
+                                     port: anything,
+                                     proto: :tls,
+                                     tls_opts: { 'cert_path' => 'path', 'private_key_path' => 'path1' }
+                                   ).once
+
+        driver.run(timeout: 1)
+      end
+
+      context 'old parameter given' do
```
Comment (repeatedly) [inline spec/fluent/plugin/in_prometheus_spec.rb]: This text is very similar to parent text. What is the different point?

File: lib/fluent/plugin/in_prometheus.rb (hunk 1 above)
Comment (ganmacs) [reply lib/fluent/plugin/in_prometheus.rb]: right.. https://github.com/fluent/fluent-plugin-prometheus/pull/136/commits/90035b68dfe8d4ddad7ce226cbb199f6e3f42d07

File: spec/fluent/plugin/in_prometheus_spec.rb (hunk 3 above)
Comment (ganmacs) [reply spec/fluent/plugin/in_prometheus_spec.rb]: fixed https://github.com/fluent/fluent-plugin-prometheus/pull/136/commits/b054290edbc7b94ff11ae62a99cc9a620da08d36

File: spec/fluent/plugin/in_prometheus_spec.rb (hunk 4 above)
Comment (ganmacs) [reply spec/fluent/plugin/in_prometheus_spec.rb]: changed descriptions https://github.com/fluent/fluent-plugin-prometheus/pull/136/commits/9edca6f388aa44d351e5ef3c7a5d5e4403da326f

File: lib/fluent/plugin/in_prometheus.rb [hunk 5]
```diff
@@ -55,6 +50,8 @@ def configure(conf)
                   nil
                 end
       @num_workers = sysconf && sysconf.workers ? sysconf.workers : 1
+      @secure = @transport_config.protocol == :tls || (@ssl && @ssl['enable'])
+      require 'openssl' if @secure
```
[the remark this answers is not in the document]
Comment (ganmacs) [reply lib/fluent/plugin/in_prometheus.rb]: fixed https://github.com/fluent/fluent-plugin-prometheus/pull/136/commits/52f0cb7c4b17888c0b7b46b65f21cfc6645424de

**Comment by ganmacs** [comment]
@repeatedly how about this?

**Comment by ganmacs** [comment]
done by https://github.com/fluent/fluent-plugin-prometheus/pull/148

**Pull request closed without merging**